tool/tsh/tsh.go

```diff
@@ -605,14 +605,24 @@ func onLogin(cf *CLIConf) {
 			tc.Logout()
 			utils.FatalError(err)
 		}
-		for _, roleName := range roleNames {
-			role, err := tc.GetRole(cf.Context, roleName)
-			if err != nil {
-				tc.Logout()
-				utils.FatalError(err)
+		// load all roles with re-used root auth client and collect
```

Comment:
Feels like the comment is unfinished "and collect ...."

Also, I would try to explain "why" instead of "what", so an explanation of why this has to be done with the root auth client would be useful.